### PR TITLE
docs: Site layout, Page templates, TOC

### LINKED ISSUE #292: Improve Docs: Create Table of Contents (TOC) component
### Description
The documentation pages (`*.Component.js` files) should generate a "(Table of) Contents" so users can easily navigate the page.

As an aside to this work, the "Properties" should move to the bottom of the page so visual examples show up first.

CORUI-6170

### PR DESCRIPTION
**Pull request by greg-a-smith**
### Description
Created documentation page templates `ComponentPage` and `MarkdownPage` to control the layout of pages.  Updated all the `*.Component.js` files to use the new page templates.  Modified the overall page layout to include a top menu bar.

Added a TOC generator (using `tocbot`) and made it responsive.

Updated responsiveness for side nav (moved the toggle button up to the top menu).

**NOTE:** There are a few follow-up items that will need to be addressed:
1. The `react-element-to-jsx-string` package that builds the JSX strings for the code samples apparently uses `displayName` so our subcomponents may need to have their display names changed to a `Parent.Sub` format.
1. The `App.scss` file is using a number of Fiori-defined colors, but not using CSS variables yet.  It should use the variables, but that is also going to require making it IE11 compatible.  That work should be done as part of #409.
1. The TOC on a `MarkdownPage` template (_e.g._ "Home" page) shows up above the `<h1>` since the `<h1>` is in the markdown file.  It would be nice to find a cleaner way to implement that.  It _looks_ okay on larger screens because it shows up in its fixed right position.

fixes #292

### CODE REVIEW

File: src/_playground/App.scss [hunk 1]
```diff
@@ -1,243 +1,303 @@
-$mediaMaxWidth: 1280px;
+$breakpoint-lg: 1280px;
```
Comment (greg-a-smith) [inline src/_playground/App.scss]: Sorry for the massive diff in here.  I flattened the class structure which is why there are so many diffs.

**Comment by bcullman** [comment]
Seeing some weirdness. When selecting a TOC item at the bottom of the list, and then navigating to another component page, the page does not return to the top.

**Comment by bcullman** [comment]
also, when selecting an item from the toc list, the title that the page jumps is getting a focus ring..see:

image

**Comment by greg-a-smith** [comment]
@bcullman Pushed changes for both issues raised.

**Comment by bcullman** [comment]
Looks good. We could arguably clean up the TOC a *bit* more on the markdown page(s) with a bit of CSS, but I would not restructure them just so the TOC falls below the `<H1>` the fact it is not under the H1 comes into play on small screens:

image

with a bit of CSS, it could look cleaner in this resolution:

image

**Comment by greg-a-smith** [comment]
@bcullman I actually found a way to leverage the heading renderer that we were already using to skip over H1 elements when the `title` prop is passed.  I think this might be a better solution yet.

**Comment by bcullman** [comment]
@greg-a-smith that is absolutely better than what I proposed. 

looks fantastic. 🚢

**Pull request merged**